### PR TITLE
Remove new_geocoder_quota feature flag

### PR DESCRIPTION
**Pull request by rafatower**
Remove all usages of `new_geocoder_quota` feature flag that was useful
when migrating quotas to redis but should be considered true in all
cases nowadays.

Related ticket: https://github.com/CartoDB/cartodb/issues/11793

### CODE REVIEW

File: services/dataservices-metrics/spec/unit/geocoder_usage_metrics_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      expect {
+        @geocoder_usage_metrics.send(
+          :check_valid_data,
+          :geocoder_internal,
+          :total_requests,
+          -42)
+      }.to raise_exception(ArgumentError, 'Invalid geocoder metric amount')
+    end
+
+    it 'validates that the amount passed cannot be zero' do
```
Comment (jgoizueta) [inline services/dataservices-metrics/spec/unit/geocoder_usage_metrics_spec.rb]: Why is zero not a valid value?

File: services/dataservices-metrics/spec/unit/service_usage_metrics_spec.rb [hunk 2]
```diff
@@ -27,4 +27,35 @@
...
+      }.to raise_exception(ArgumentError, 'Invalid metric amount')
+    end
+
+    it 'validates that the amount passed cannot be negative' do
+      expect {
+        @usage_metrics.send(:assert_valid_amount, -42)
+      }.to raise_exception(ArgumentError, 'Invalid metric amount')
+    end
+
+    it 'validates that the amount passed cannot be zero' do
```
Comment (jgoizueta) [inline services/dataservices-metrics/spec/unit/service_usage_metrics_spec.rb]: ditto: why cannot be zero?

File: lib/tasks/user_overquota.rake [hunk 3]
```diff
@@ -34,11 +34,7 @@ namespace :cartodb do
         number_cycles.times do |cycle|
           date_from = billing_date << cycle
           date_to = billing_date << (cycle-1)
-          if args[:redis]
-            geocodings = user.get_new_system_geocoding_calls(from: date_from, to: date_to)
-          else
-            geocodings = user.get_db_system_geocoding_calls(date_from, date_to)
-          end
+          geocodings = user.get_new_system_geocoding_calls(from: date_from, to: date_to)
```
Comment (jgoizueta) [inline lib/tasks/user_overquota.rake]: I guess you're not removing the :redis argument from the rake task to maintain compatibility with any scripts using this task.

**Comment by rafatower** [comment]
Arguably passing a zero to increment a counter is a programming error on the client. Other than that it would produce an undesired request to the redis storage. But as I cannot guarantee 100% that the client code is guarded against that I'll relax that condition on the service class. Thanks for the review!

**Pull request merged**